### PR TITLE
doc! update to latest docs changes and rename to Cawaena

### PR DESCRIPTION
**Pull request by kristifidani**
# Pull Request Template

## Motivation and Context

Some changes to the public docs were overridden while attempting to rename from Cryptpay to Cawaena. Therefore, we need to bring the changes back.

## Summary

- Reverted docs to reference the lastest ones from the migration repo: [ETO-GRUPPE-TECHNOLOGIES-GmbH/Cryptpay-SDK (github.com)](https://github.com/ETO-GRUPPE-TECHNOLOGIES-GmbH/Cryptpay-SDK)
- Build docs for Java and Rust. Updated some links in the Api references.
- Kept some of the renaming from `Cryptpay` to `Cawaena`. Of course, there are places where we are still using cryptpay but then we will gradually have to change these as they are mostly code snippets, docs references, files etc which are not as straightforward to do with `replace all`.

## Checklist

Please ensure that your PR meets the following requirements:

### General

- [ ] Code follows project style and guidelines.
- [ ] No redundant or duplicate code.
- [ ] No added new dependencies without clear justification and approval.

### Documentation

- [ ] Added/updated relevant documentation (e.g., README, inline comments).
- [ ] Added a CHANGELOG entry for major changes.

### Testing

- [ ] Added relevant test cases, leveraging tools like `rstest` or similar for reusable tests.
- [ ] Tests pass locally.

### Build & CI/CD

- [ ] Confirmed no CI/CD pipeline issues.
- [ ] Updated build scripts where necessary.

## Comments to Reviewer

Please provide any comments or points of consideration for the reviewers, such as specific areas of focus, potential edge cases, or context that would aid the review process.

### Reviewer Responsibilities

- [ ] Code adheres to style and guidelines.
- [ ] All tests are appropriately covered and they pass.
- [ ] Consider performance and/or security impacts.
- [ ] Constructive feedback is provided.

### CODE REVIEW

**Review by antbern — APPROVED** [review]
Let's merge so we can continue updating these 💯

File: .github/workflows/github-pages.yml [hunk 1]
```diff
@@ -52,17 +52,32 @@ jobs:
         run: wasm-pack build --dev --scope eto
         working-directory: sdk/bindings/wasm
 
+      - name: Build Rust docs
+        shell: bash
+        run: |
+          cargo doc --no-deps --all-features --lib --target-dir rust-docs
+          mv rust-docs/ ./docs/
```
Comment (antbern) [inline .github/workflows/github-pages.yml:59]:
```suggestion
          cargo doc --no-deps --all-features --lib
          mkdir ./docs/rust-docs/docs/
          mv ../target/doc/* ./docs/rust-docs/docs/
```

**Review by antbern — CHANGES_REQUESTED** [review]

File: .github/workflows/github-pages.yml [hunk 2]
```diff
@@ -52,17 +52,33 @@ jobs:
         run: wasm-pack build --dev --scope eto
         working-directory: sdk/bindings/wasm
 
+      - name: Build Rust docs
+        shell: bash
+        run: |
+          cargo doc --no-deps --all-features --lib
+          mkdir ./docs/rust-docs/docs/
```
Comment (antbern) [inline .github/workflows/github-pages.yml:59]:
```suggestion
          mkdir -p ./docs/rust-docs/docs/
```

File: .github/workflows/github-pages.yml [hunk 3]
```diff
@@ -52,17 +52,33 @@ jobs:
         run: wasm-pack build --dev --scope eto
         working-directory: sdk/bindings/wasm
 
+      - name: Build Rust docs
+        shell: bash
+        run: |
+          cargo doc --no-deps --all-features --lib
+          mkdir -p ./docs/rust-docs/docs/
+          mv ../target/doc/* ./docs/rust-docs/docs/
```
Comment (antbern) [inline .github/workflows/github-pages.yml:60]:
```suggestion
          mkdir -p ./docs/rust-docs/doc/
          mv ../target/doc/* ./docs/rust-docs/doc/
```

**Review by antbern — APPROVED** [review]

**Review by antbern — APPROVED** [review]
